src/js/components/Select/Select.js

```diff
@@ -267,11 +267,27 @@ const Select = forwardRef(
 
     // element to show, trumps inputValue
     const selectValue = useMemo(() => {
-      if (valueLabel) return valueLabel;
-      if (React.isValidElement(value)) return value; // deprecated
+      if (valueLabel instanceof Function) {
+        if (value) return valueLabel(value);
+      } else if (valueLabel) return valueLabel;
+      else if (React.isValidElement(value)) return value; // deprecated
       return undefined;
     }, [value, valueLabel]);
 
+    // if labelKey is a function and no valueLabel is defined
+    // we should use the labelKey function to display the
+    // selected value
+    const displayLabelKey = useMemo(() => {
+      if (
+        !selectValue &&
+        optionIndexesInValue.length === 1 &&
+        typeof applyKey(allOptions[optionIndexesInValue[0]], labelKey) ===
```

Comment:
Seems like we are calculating this twice. Once to get the typeof and a second time when we return. Is there a chance to pull this out as a variable so we just run the function once?

Reply:
Yeah makes sense, I fixed this in latest commit